sktime/transformations/series/dropna.py

```diff
@@ -0,0 +1,210 @@
+
+    def _fit(self, X, y=None):
+        """Fit transformer to X and y.
+
+        private _fit containing the core logic, called from fit
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            if self.get_tag("univariate-only")==True:
+                guaranteed to have a single column
+            if self.get_tag("univariate-only")==False: no restrictions apply
+        y : None, present only for interface compatibility
+
+        Returns
+        -------
+        self: reference to self
+        """
+        self.dropped_index_values_ = None
+
+        agg_axis = 1 - self._axis
+
+        mask = None
+        if self._how == "any":
+            mask = X.isna().any(axis=agg_axis)
+        elif self._how == "all":
+            mask = X.isna().all(axis=agg_axis)
+        elif isinstance(self._thresh, int):
+            mask = X.count(axis=agg_axis) < self._thresh
+        elif isinstance(self._thresh, float):
+            mask = X.count(axis=agg_axis).div(X.shape[agg_axis]) < self._thresh
+
+        if mask is not None:
+            self.dropped_index_values_ = mask.index[mask].to_list()
+        else:
+            self.dropped_index_values_ = None
+
+        return self
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            if self.get_tag("univariate-only")==True:
+                guaranteed to have a single column
+            if self.get_tag("univariate-only")==False: no restrictions apply
+        y : None, present only for interface compatibility
+
+        Returns
+        -------
+        transformed version of X
+        """
+        dropped_index_values = self.dropped_index_values_
+        axis = self._axis
+
+        if dropped_index_values is not None:
+            return X.drop(labels=dropped_index_values, axis=axis)
+        else:
+            return X
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for transformers.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        params = [
+            {"axis": 0, "how": "any", "thresh": None},
```

Comment:
that's a lot of parameter combinations - looks exhaustive.
Might be worth checking the runtime and reducing this if it takes longer than, say, 20 sec.